lib/mutant/mutator/node/define.rb

```diff
@@ -13,11 +13,39 @@ class Define < self
         # @api private
         def dispatch
           emit_arguments_mutations
+          emit_optarg_mutations
           emit_body(N_RAISE)
           emit_body(nil)
           emit_body_mutations if body
         end
 
+        # Emit mutation with optional arguments as assignments in method
+        #
+        # @return [undefined]
+        #
+        # @api private
+        def emit_optarg_mutations
+          optargs, args = arguments.children.partition do |child|
+            child.type.equal?(:optarg)
+          end
+
+          return if optargs.empty?
+
+          lvasgns = optargs.map do |optarg|
+            s(:lvasgn, *optarg.children)
+          end
+
+          body_children = [*lvasgns, body].compact
+
+          new_body = if body_children.one?
+            body_children.first
+          else
+            s(:begin, *body_children)
+          end
```

Comment:
I add this conditional to handle for if there is one or multiple nodes in the body.  If there is only one then we don't want to wrap it with `s(:begin, ...)` because it looks like unparser then finds the output to be different in the `Mutant::Meta::Example`.